src/elfpy/utils/data.py

```diff
@@ -18,23 +18,45 @@ def format_trades(analysis_dict):
     trades["fee_in_usd"] = trades.fee
     trades["fee_in_bps"] = trades.fee / trades.out_without_fee * 100 * 100
     base_asset_liquidity_usd = trades.base_asset_reserves * trades.base_asset_price
-    token_asset_liquidity_usd = (
-        trades.token_asset_reserves * trades.base_asset_price * trades.spot_price
-    )
+    token_asset_liquidity_usd = trades.token_asset_reserves * trades.base_asset_price * trades.spot_price
     trades["total_liquidity_usd"] = base_asset_liquidity_usd + token_asset_liquidity_usd
     trades["trade_volume_usd"] = trades.out_with_fee
     # calculate percent change in spot price since the first spot price (after first trade, kinda weird)
-    trades["price_rate_of_change"] = (
+    trades["price_total_return"] = (
         trades.loc[:, "spot_price"] / trades.loc[0, "spot_price"] - 1
-    )  # rescales spot_price_rate_of_change to equal init_share_price for the first value, for comparison
+    )  # rescales price_total_return to equal init_share_price for the first value, for comparison
     trades["price_total_return_scaled_to_share_price"] = (
-        trades.price_rate_of_change + 1
+        trades.price_total_return + 1
     ) * trades.init_share_price  # this is APR (does not include compounding)
+    trades["lp_return"] = trades.fee / trades.total_liquidity_usd
+    trades["lp_total_return"] = 0
+    trades["share_price_total_return"] = 0
+    for run in trades.run_number.unique():
+        trades.loc[trades.run_number == run, "lp_total_return"] = trades.loc[
+            trades.run_number == run, "lp_return"
+        ].cumsum()
+        trades.loc[trades.run_number == run, "lp_total_return_scaled_to_share_price"] = (
+            trades.lp_total_return + 1
+        ) * trades.init_share_price  # this is APR (does not include compounding)
+        trades.loc[trades.run_number == run, "share_price_total_return"] = (
+            trades.loc[trades.run_number == run, "share_price"]
+            / trades.loc[trades.run_number == run, "share_price"].iloc[0]
+            - 1
+        )
+    trades["lp_total_return_percent"] = trades.lp_total_return * 100
+    trades["price_total_return_percent"] = trades.price_total_return * 100
+    trades["share_price_total_return_percent"] = trades.share_price_total_return * 100
+    scale = 365 / ( trades["day"] + 1 )
+    trades["lp_total_return_percent_annualized"]           = scale * trades["lp_total_return_percent"]
+    trades["price_total_return_percent_annualized"]        = scale * trades["price_total_return_percent"]
+    trades["share_price_total_return_percent_annualized"]  = scale * trades["share_price_total_return_percent"]
+
     # create explicit column that increments per trade
     trades = trades.reset_index()
-    # aggregate trades over days
+
+    ### STATS AGGREGATED BY DAY ###
     keep_columns = [
-        "model_name",
+        "model_name" if trades.model_name.nunique() > trades.scenario_name.nunique() else "scenario_name",
```

Comment:
Can you add a comment noting the difference between "model_name" and "scenario_name"?